### ISSUE TITLE
OnionShare doesn't run on macOS M1 (ARM) Apple Silicon Builds

### ISSUE THREAD

**Issue by agnosticlines** [onionshare/onionshare#1414]
Hey there,

I'm unable to run onion share even through rosetta as it gives the following error:

```
Traceback (most recent call last):
  File "OnionShare.app/Contents/Resources/Support/Python/Resources/lib/python3.8/runpy.py", line 111, in _get_module_details
    __import__(pkg_name)
  File "OnionShare.app/Contents/Resources/app/onionshare/__init__.py", line 27, in <module>
    import psutil
  File "OnionShare.app/Contents/Resources/app_packages/psutil/__init__.py", line 121, in <module>
    from . import _psosx as _psplatform
  File "OnionShare.app/Contents/Resources/app_packages/psutil/_psosx.py", line 15, in <module>
    from . import _psutil_osx as cext
ImportError: dlopen(/Applications/OnionShare.app/Contents/Resources/app_packages/psutil/_psutil_osx.cpython-38-darwin.so, 2): no suitable image found.  Did find:
	/Applications/OnionShare.app/Contents/Resources/app_packages/psutil/_psutil_osx.cpython-38-darwin.so: mach-o, but wrong architecture
	/Applications/OnionShare.app/Contents/Resources/app_packages/psutil/_psutil_osx.cpython-38-darwin.so: mach-o, but wrong architecture
```

It'd be really nice to have a prebuilt package compatible with M1 Macs :-)

Thanks!

**Comment by hugooalmeidaa** [comment]
I'm having the same issue, tried to reinstall a lot of time and didn't work too.

**Comment by micahflee** [comment]
Good to know! I don't have an M1 Mac myself, but I'd love to support this.

**Comment by agnosticlines** [comment]
So I think this can be pretty easily fixed by switching out the app_resources/* with M1 python package eggs, I'll try today and let you know how I get on

**Comment by hugooalmeidaa** [comment]
Open with Roseta worked!! Thank you!!

**Comment by hugooalmeidaa** [comment]
As I see in the web, Roseta is just a solution for now, it was created to developers get some time to update the apps for the new architecture on Mac newer hardware. So I believe in the future will be request a update in Onion Share app. If anyone would like to talk more about how we could do that, feel free to contact me. I'm free to help

**Comment by harrisondesbrosses** [comment]
I made a duplicate thread for this issue, but I did try to locate a similar thread beforehand (to no avail). Thank you all for this. 

Can anyone explain the Rosetta method to me? I’d like to use a temporary fix in the meantime.

**Comment by harrisondesbrosses** [comment]
I would like to request proper support for silicon macs in the future. How might I do that?

**Comment by harrisondesbrosses** [comment]
Additionally, does anyone know if running OS via Rosetta is a security issue?

**Comment by mig5** [comment]
I believe this is now fixed in 2.6.1, should be able to run on M1.
